trade-module/src/main/java/io/edpn/backend/trade/application/port/outgoing/marketdatum/CreateWhenNotExistsLatestMarketDatumPort.java

```diff
@@ -0,0 +1,10 @@
+package io.edpn.backend.trade.application.port.outgoing.marketdatum;
+
+import io.edpn.backend.trade.application.domain.MarketDatum;
+
+import java.util.UUID;
+
+public interface CreateWhenNotExistsLatestMarketDatumPort {
+
+    void createWhenNotExists(UUID stationId, MarketDatum marketDatum);
```

Comment:
Rename port to something like:
createOrUpdateExisting

This differs fundamentally from how the full MarketDatum table operates and this method should indicate that.